File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/documents/DocumentCaseContainers.scala
package dk.openesdh.repo.webscripts.documents

import com.typesafe.scalalogging.slf4j.StrictLogging
import dk.openesdh.repo.services.cases.CaseService
import dk.openesdh.repo.services.documents.DocumentService
import org.alfresco.service.cmr.repository.{InvalidNodeRefException, NodeRef}
import org.springframework.extensions.webscripts.{Cache, DeclarativeWebScript, Status, WebScriptRequest}
import scala.collection.JavaConversions._
import scala.collection.mutable._

/**
 * @author lanre
 */
class DocumentCaseContainers(val caseService: CaseService, val documentService: DocumentService) extends DeclarativeWebScript with StrictLogging{
   protected override def executeImpl(req: WebScriptRequest, status: Status, cache: Cache) :  java.util.Map[String, Object] = {

     val templateArgs = req.getServiceMatch.getTemplateVars

     try {
       val storeType: String = templateArgs.get ("store_type")
       val storeId: String = templateArgs.get ("store_id")
       val id: String = templateArgs.get ("id")
       val docNodeRefStr = s"$storeType://$storeId/$id"
       val documentNode: NodeRef = new NodeRef (docNodeRefStr)
       val caseNodeRef = documentService.getCaseNodeRef(documentNode)
       val caseDocumentNodeRef: Object = caseService.getDocumentsFolder(caseNodeRef)

       val model: Map[java.lang.String, java.lang.Object] = new HashMap[java.lang.String, java.lang.Object] ()
       model.put("caseNodeRef", caseNodeRef)
       model.put("caseDocumentNodeRef", caseDocumentNodeRef)

       model
     }
     catch {
       case inre: InvalidNodeRefException => {
         //logger.error("", inre)
         null
       }
     }
   }

}

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/documents/DocumentCaseContainers.java
package dk.openesdh.repo.webscripts.documents;

import java.util.HashMap;
import java.util.Map;

import org.alfresco.service.cmr.repository.InvalidNodeRefException;
import org.alfresco.service.cmr.repository.NodeRef;
import org.apache.commons.lang3.StringUtils;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.springframework.extensions.webscripts.Cache;
import org.springframework.extensions.webscripts.DeclarativeWebScript;
import org.springframework.extensions.webscripts.Status;
import org.springframework.extensions.webscripts.WebScriptRequest;

import dk.openesdh.repo.services.cases.CaseService;
import dk.openesdh.repo.services.documents.DocumentService;

/**
 * @author Lanre.
 */
public class DocumentCaseContainers extends DeclarativeWebScript {
    private static Log logger = LogFactory.getLog(DocumentCaseContainers.class);
    private CaseService caseService;
    private DocumentService documentService;

    @Override
    protected Map<String, Object> executeImpl(WebScriptRequest req, Status status, Cache cache){
        Map<String, String> templateArgs = req.getServiceMatch().getTemplateVars();

        String storeType = templateArgs.get("store_type");
        String storeId = templateArgs.get("store_id");
        String id = templateArgs.get("id");
        String caseId = templateArgs.get("caseId");
        NodeRef documentNode = (StringUtils.isNotEmpty(storeId)) ? new NodeRef(storeType, storeId, id) : null;
        NodeRef caseNodeRef;

        Map<String, Object> model = new HashMap<String, Object>();
        try{
            if(StringUtils.isNotEmpty(caseId))
                caseNodeRef = caseService.getCaseById(caseId);
            else
                caseNodeRef = caseService.getParentCase(documentNode);
            NodeRef caseDocumentNodeRef  = caseService.getDocumentsFolder(caseNodeRef);

            model.put("caseNodeRef", caseNodeRef.toString());
            model.put("caseDocumentNodeRef", caseDocumentNodeRef.toString());
        }
        catch (InvalidNodeRefException inre) {
            logger.error("Unable to get case document containers due to the following error: "+ inre.getMessage());
        }

        return model;
    }

    public void setCaseService(CaseService caseService) {
        this.caseService = caseService;
    }

    public void setDocumentService(DocumentService documentService) {
        this.documentService = documentService;
    }
}
